param/parameterized.py

```diff
@@ -3358,6 +3375,11 @@ def __param_inheritance(mcs, param_name, param):
                 else:
                     setattr(param, slot, default_val)
 
+            # Avoid crosstalk between mutable slot values in different Parameter objects
+            v = getattr(param, slot)
+            if (not getattr(param, "constant", False) and is_mutable(v)):
```

Comment:
Why dealing with `constant` here?

Reply:
Oh ok I see, that had to do with the failing test. Why not just doing like in https://github.com/holoviz/param/pull/826/files#diff-c8fc4fb68012b66fbf42ad9e0b4714fd96a3a72b83558bc543130d78bd622eb3R1711 and not setting the parameter slot when the slot is `'default'`?

Reply:
Because I didn't think of that. :-) I think that's the right approach, at least for now; copying `default` is a separate issue.

Reply:
And I see now your related comment:

> I currently check for default only when copying the Parameter object, not during inheritance. It might also be appropriate to skip shallow-copying for for inheritance as well, but I haven't found an example where that would come up. Probably worth doing?

I'd be in favor not to touch `default` in this PR.

Reply:
Ok, done and pushed.